core/src/scene/styleParam.h

```diff
@@ -141,8 +143,20 @@ struct StyleParam {
         }
     };
 
-    using Value = variant<Undefined, none_type, bool, float, uint32_t, std::string, glm::vec2,
-                            Width, LabelProperty::Placement, LabelProperty::Anchors>;
+
+    struct TextSource {
+        std::vector<std::string> keys;
+        bool operator==(const TextSource& _other) const {
+            if (keys.size() != _other.keys.size()) { return false; }
```

Comment:
I think that `keys == _other.keys` should be sufficient here and more succinct. 
See operator the doc here http://en.cppreference.com/w/cpp/container/vector/operator_cmp, the size is the first element compared before the content.